spi/src/main/java/org/opensearch/security/spi/resources/client/ResourceSharingClient.java

```diff
@@ -47,11 +45,4 @@ public interface ResourceSharingClient {
      * @param listener       The listener to be notified with the updated ResourceSharing document.
      */
     void revoke(String resourceId, String resourceIndex, ShareWith target, ActionListener<ResourceSharing> listener);
-
-    /**
-     * Lists resourceIds of all shareable resources accessible by the current user.
-     * @param resourceIndex The index containing the resources.
-     * @param listener The listener to be notified with the set of accessible resources.
-     */
-    void getAccessibleResourceIds(String resourceIndex, ActionListener<Set<String>> listener);
```

Comment:
ML uses modelGroupIds to filter search for models. We need to keep this here until we support hierarchy.